src/hooks/useAllPairsWithLiquidityAndMaximumApyAndStakingIndicator.tsx

```diff
@@ -1,13 +1,35 @@
-import { Pair, CurrencyAmount, Percent, Token, KpiToken } from '@swapr/sdk'
+import { Pair, CurrencyAmount, Percent, Token, KpiToken, ZERO, BigintIsh, JSBI } from '@swapr/sdk'
 import { useMemo } from 'react'
 import { PairsFilterType } from '../components/Pool/ListFilter'
-import { useAggregatedByToken0PairComparator } from '../components/SearchModal/sorting'
+import { AggregatedPairs } from '../components/Pool/PairsList'
+import { LIQUIDITY_SORTING_TYPES } from '../constants'
 import { getBestApyPairCampaign } from '../utils/liquidityMining'
 import { useAllPairsWithNonExpiredLiquidityMiningCampaignsAndLiquidityAndStakingIndicator } from './useAllPairsWithNonExpiredLiquidityMiningCampaignsAndLiquidityAndStakingIndicator'
 
+const liquidityPairSorter: {
+  [sortBy: string]: (pairA: AggregatedPairs, pairB: AggregatedPairs) => number
+} = {
+  [LIQUIDITY_SORTING_TYPES.TVL]: (
+    pairA: { liquidityUSD: CurrencyAmount },
+    pairB: { liquidityUSD: CurrencyAmount }
+  ): number => {
+    if (pairA.liquidityUSD.equalTo(pairB.liquidityUSD)) return 0
+    return pairA.liquidityUSD.lessThan(pairB.liquidityUSD) ? 1 : -1
+  },
+  [LIQUIDITY_SORTING_TYPES.APY]: (pairA: { maximumApy: Percent }, pairB: { maximumApy: Percent }): number => {
+    if (pairA.maximumApy.equalTo(pairB.maximumApy)) return 0
+    return pairA.maximumApy.lessThan(pairB.maximumApy) ? 1 : -1
+  },
+  [LIQUIDITY_SORTING_TYPES.NEW]: (pairA: { startsAt?: BigintIsh }, pairB: { startsAt?: BigintIsh }): number => {
+    if (JSBI.EQ(pairA.startsAt, pairB.startsAt)) return 0
+    return JSBI.LT(pairA.startsAt, pairB.startsAt) ? 1 : -1
+  },
+}
+
```

Comment:
we should also this one like you did with previous sorter inside the liquidtyMining utils folder

Reply:
We are using it only in this file, I think there's no current advantage to move it.